pypika/terms.py

```diff
@@ -342,7 +342,8 @@ def get_value_sql(self, **kwargs: Any) -> str:
         if isinstance(self.value, Term):
             return self.value.get_sql(**kwargs)
         if isinstance(self.value, Enum):
-            return self.value.value
+            self.value = self.value.value
```

Comment:
I think it would be best if the value was reset here. Maybe this method be refactored so you can just pass in the new value, or even just instantiate another ValueWrapper with the value? e.g.
```python
return ValueWrapper(self.value.value, self.alias).get_value_sql(**kwargs)
```

What do you think>

Reply:
I agree with @mikeengland. Also that introduces a side effect to get_sql, which imo could be really tricky to debug, given nobody would expect get_sql to be actually modifying the object instance.

Reply:
@mikeengland I agree. Initially I wanted to pass it as an argument to `get_value_sql` but that's not possible without refactoring like you said.

I have updated the PR with tests and some refactoring of `get_value_sql`. The idea is to be able to pass a `_value` argument when you want to call `get_value_sql` recursively. The idea is that `get_value_sql` should include the logic for quoting base types like `int`, `bool`, `str` and such. For other, more complex types (`date`, `Enum`, etc.) it handles the conversion one of the base types and then calls itself with the converted type. As an example I have refactored the `date` conversion the same way `Enum`s are handled, which basically gets rid of the duplicate logic for `format_quotes` which was already handle for `str`.

If you don't like it, I could also implement your suggestion of instantiating another `ValueWrapper`

Reply:
There is only one 'hack' if you want to call it that, and that is the introduction of an `UNSET` type (which is just an `object`), so that `None` is handled properly.